hazelcast/src/main/java/com/hazelcast/nio/serialization/GenericRecord.java

```diff
@@ -43,23 +45,23 @@
  *
  *             return null;
  *         });
- * </pre>
- * Another example with EntryProcessor to demonstrate how to read, modify and set back to the map:
- * <pre>
+ * }</pre>
+ * Another example with EntryProcessor to demonstrate how to read, modify and set the value back to the map:
+ * <pre>{@code
  * map.executeOnKey("key", (EntryProcessor<Object, Object, Object>) entry -> {
  *             GenericRecord genericRecord = (GenericRecord) entry.getValue();
  *             GenericRecord modifiedGenericRecord = genericRecord.cloneWithBuilder()
  *                     .setInt("age",22).build();
  *             entry.setValue(modifiedGenericRecord);
  *             return null;
  *         });
- * </pre>
+ * }</pre>
  * <p>
- * GenericRecord also allows to read from a cluster without having the classes on the client side.
+ * GenericRecord also allows reading from the cluster without having the classes on the client side.
  * For {@link Portable}, when {@link PortableFactory} is not provided in the config at the start,
  * a {@link HazelcastSerializationException} was thrown stating that a factory could not be found. Starting from 4.1,
- * the objects will be returned as {@link GenericRecord}. This way, the clients can read and write the objects back to
- * the cluster without needing the classes of the domain objects on the classpath.
+ * the objects will be returned as {@link GenericRecord}. This way, the clients can read and write objects back to
+ * the cluster without the need that the domain classes are on the classpath.
```

Comment:
```suggestion
 * the cluster without the need of having the domain classes on the classpath.
```

or 

```suggestion
 * the cluster without needing the domain classes on the classpath.
```

I prefer the latter one since it's simpler